### ISSUE TITLE
Hires Fix + Inpainting => color change

### ISSUE THREAD

**Issue by imperator-maximus** [comfyanonymous/ComfyUI#2827]
Hello,

I tried the hi-res fix example using Inpainting:
https://comfyanonymous.github.io/ComfyUI_examples/2_pass_txt2img/

The inpainting result looks good but original pixels outside mask are pixelated which also makes sense because I downscaled 1024 image to 512 made the Inpainting and used Latent Upscale coming back to 1024. The whole workflow looks very similar but I used normal non inpainting model + Inpainting VAE Encode + non inpainting model for second sampler. 
This is mask and result:

![clipspace-mask-899862 5](https://github.com/comfyanonymous/ComfyUI/assets/4406586/578e70fd-8586-4b76-ab33-942dda33a9ac)

![ComfyUI_temp_siotg_00063_](https://github.com/comfyanonymous/ComfyUI/assets/4406586/96c76d5e-e0e5-44d7-8883-1b57a3fb74ec)

To solve the issue I added LatentCompositionMasked node which might be same as "apply_overlay" parameter in Diffuser project and got this result:
![ComfyUI_temp_siotg_00062_](https://github.com/comfyanonymous/ComfyUI/assets/4406586/bcc37155-334c-48ab-b28e-f016783c1c00)

This looks not so bad (beside the masked edges at the lower left but my mask is quite large - so this is just as it is).
Now comes the problem the whole image gets an drastic color change here putting it in original context:

![Screenshot 2024-02-18 170416](https://github.com/comfyanonymous/ComfyUI/assets/4406586/7ca98013-dc8a-4d39-8f60-a246507e18fa)

This is a  known problem other servers have but overall it is solved more than a year ago. So I am wondering where there isn't any example showing hires fix+ Inpainting on 1.x at all in whole internet. Thank you.

**Comment by HelloClyde** [comment]
You might need to use the ImageCompositeMasked node to overlay the original image on the non-masked parts, and the mask could use some feathering.

**Comment by imperator-maximus** [comment]
> You might need to use the ImageCompositeMasked node to overlay the original image on the non-masked parts, and the mask could use some feathering.

yes that is what I tried first but results had been more worse
![ComfyUI_temp_siotg_00064_](https://github.com/comfyanonymous/ComfyUI/assets/4406586/819fc7e3-8f2b-4cbe-85c7-94eb7e84f6a2)

**Comment by HelloClyde** [comment]
![image](https://github.com/comfyanonymous/ComfyUI/assets/12826558/c50388d7-ec75-44de-a2a0-bb341d9e36f7)
I guess it might have something to do with the model capacity and VAE? I tried it and it wasn't very exaggerated.

**Comment by HelloClyde** [comment]
My understanding is that both Hires Fix and inpainting are generated through diffusion models, and even with the use of the LatentCompositionMasked node, the final result will definitely be different from the original image (in terms of color, details, etc.). That's why the last step requires the use of ImageCompositeMasked to overlay the non-masked parts. ImageCompositeMasked is purely an image overlay (similar to layer overlay in Photoshop), which ensures that the non-masked parts are exactly the same as the original image.

**Comment by imperator-maximus** [comment]
ok I made more tests and removed HiResFix but problem is still the same just using the standard Inpainting example: edges around the mask have color change or edges at the whole image are wrong. So I guess 1.5+Inpainting just does not work.

**Comment by DaviSMoura** [comment]
@imperator-maximus any news in fixing it?

**Comment by imperator-maximus** [comment]
> @imperator-maximus any news in fixing it?

yes these nodes do this job! Color change is even better than SDXL standard inpainting (it darkens the color a bit at SDXL)
![image](https://github.com/comfyanonymous/ComfyUI/assets/4406586/5851deab-0a71-4805-8cb3-4ba36784fbb6)

**Comment by imperator-maximus** [comment]
1.5 seems to work fine now. Will check SDXL options later.
